### PR TITLE
Add storage features

### LINKED ISSUE #68: Storage features
Store all sleep, activity, diet, for each, goals

### PR DESCRIPTION
**Pull request by skylee03**
- Closes #68 

Since serialized objects are stored, manual modification of the file is not supported.

### CODE REVIEW

**Review by yicheng-toh — APPROVED** [review]
LGTM

Should we let the user do the saves or should the entries be saved after execution of commands?

**Comment by skylee03** [comment]
> LGTM
> 
> Should we let the user do the saves or should the entries be saved after execution of commands?

Data is only saved when executing the `save` command or the `bye` command. I'm not implementing an autosave feature because I don't want to have too many unnecessary file IO. Although the resources consumed on file IO are insignificant given our limited data size, for larger projects we should strictly avoid doing so.

**Comment by skylee03** [comment]
Currently, as long as the program exits normally using the `bye` command, the data will be automatically saved.

However, in the case of other exits, data is not saved automatically. This problem can be solved using `addShutdownHook`.

File: src/main/java/athleticli/storage/Config.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package athleticli.storage;
+
+/**
+ * Defines string literals or configurations used for file storage.
+ */
+public class Config {
+    public static final String PATH_SAVE = "./data/athleticli.txt";
```
Comment (skylee03) [inline src/main/java/athleticli/storage/Config.java:7]: This is not a plain text file, so the `.txt` extension should be avoided.
```suggestion
    public static final String PATH_SAVE = "./data/athleticli.bin";
```

File: src/main/java/athleticli/AthletiCLI.java [hunk 2]
```diff
@@ -25,7 +26,7 @@ public class AthletiCLI {
      */
     public AthletiCLI() {
         ui = new Ui();
-        data = new Data();
+        data = Storage.load();
```
Comment (skylee03) [inline src/main/java/athleticli/AthletiCLI.java:29]: Should we give the user feedback when load fails?

**Pull request merged**